client/src/modules/cash/cash.html

```diff
@@ -67,129 +67,141 @@
                   max-date="CashCtrl.timestamp">
                 </bh-date-editor>
 
-                <!-- currency selection -->
-                <bh-currency-select
-                  currency-id="CashCtrl.Payment.details.currency_id"
-                  validation-trigger="CashVoucherForm.$submitted"
-                  on-change="CashCtrl.Payment.setCurrency(currency)"
-                  disable-ids="CashCtrl.disabledCurrencyIds"
-                  cashbox-id="CashCtrl.cashbox.id">
-                </bh-currency-select>
-
-                <!-- determine if we are creating a caution or an invoice payment -->
-                <div class="radio" ng-class="{ 'has-error' : CashVoucherForm.$submitted && CashVoucherForm.type.$invalid }">
-                  <p><strong class="control-label" translate>FORM.LABELS.TYPE</strong></p>
-                  <label class="radio-inline">
-                    <input name="type" type="radio" ng-model="CashCtrl.Payment.details.is_caution" ng-value="1" ng-change="CashCtrl.Payment.setCautionType(CashCtrl.Payment.details.is_caution)" data-caution-option="1" required>
-                    <span translate>FORM.LABELS.CAUTION</span>
-                  </label>
-                  <label class="radio-inline">
-                    <input name="type" type="radio" ng-model="CashCtrl.Payment.details.is_caution" ng-value="0" ng-change="CashCtrl.Payment.setCautionType(CashCtrl.Payment.details.is_caution)" data-caution-option="0" required>
-                    <span translate>FORM.LABELS.INVOICE</span>
-                  </label>
-
-                  <div class="help-block" ng-messages="CashVoucherForm.type.$error" ng-show="CashVoucherForm.$submitted">
-                    <div ng-messages-include="modules/templates/messages.tmpl.html"></div>
+                <!-- allowed for cash payment -->
+                <section ng-if="!CashCtrl.Payment.messages.isNonCashPatient">
```

Comment:
This section should include the date input  - there is no reason to see it if we are hiding everything else.

Reply:
Instead of hiding everything, I propose that you wrap it a
```html
<fieldset ng-disabled="CashCtrl.Payment.messages.isNonCashPatient">
```
You can then put a small alert on top of this saying that the patient cannot pay here - they are covered by Debtor Group X.  I would also make the alert red since it will be blocking all further activity.